pilot/pkg/security/authz/policy/v1beta1/v1beta1.go

```diff
@@ -71,9 +71,19 @@ func (g *v1beta1Generator) generatePolicy(policies []model.AuthorizationPolicyCo
 			if rule == nil {
 				continue
 			}
-			m := authzModel.New(g.trustDomainBundle, rule)
+			m, err := authzModel.New(rule)
+			if err != nil {
+				rbacLog.Errorf("failed to construct model: %v", err)
+				continue
+			}
+			m.MigrateTrustDomain(g.trustDomainBundle)
```

Comment:
MigrateTrustDomain() is not the original logic. Is it a bug fix?

Reply:
It is the original logic for the trust domain alias support (added by Phillip before), I extracted the logic and put it in a separate function to make it easier to reuse. This PR only includes code refactor and no bug fixes.